File: 1158.c
//닭이 없으면? 꿩을 먹으면 된다!!!
#include <stdio.h>
int table[5000]={1};
int checkTable(){
    int c=0;
    int length=0;
    for (int i=0; table[i]!=0; i++) length++;
    for (int i=0; table[i]!=0; i++){
        if (!table[i]) c+=1;
        else c+=0;
    }
    if (c==length) return 0;
    else return 1;
}
int main(){
    int n,k;
    scanf("%d %d", &n, &k);
    int i=1;
    int copy_k=k-1;
    while(checkTable()){
        if (copy_k==0 && table[i-1]){
            table[i-1]=0;
            copy_k=k-1;
            printf(checkTable() ? "%d, " : "%d>", i);
            printf("%d",i);
            i=i!=n ? i+1 : 1;
            continue;
        }
        if (table[i-1]) copy_k-=1;
        i=i!=n ? i+1 : 1;
    }
    printf("\n");
}

File: 1158.py
#웰노운
#(7,3) -> (1) (2) 3 (4) (5) 6 (7) (1) 2 (4) (5) 7 (1) (4) 5 (1) (4) 1 (4) (4) 4
def checkTable(table):
    c=0
    length=len(table)
    for element in table: c+=1 if not(element) else 0
    return False if c==length else True
n,k=map(int,input().split())
table=[True for _ in range(n)] #카운트를 줄이는 용도로
#리스트 참조는 (i-1)번째로.
#True이면 1 줄이고, False이면 유지
#만약 한바퀴를 다 돌았다면 다시 i를 1로 복귀
i=1
copy_k=k-1
how_many_times=0
print("<",end='')
while (checkTable(table)):
    if copy_k==0 and table[i-1]:
        table[i-1]=False
        copy_k=k-1
        print(f"{i}, " if how_many_times<6 else f"{i}>",end='')
        i=i+1 if i!=n else 1
        how_many_times+=1
        continue 
    if table[i-1]: copy_k-=1
    i=i+1 if i!=n else 1
print()